oximeter/collector/src/lib.rs

```diff
@@ -202,10 +247,32 @@ impl Oximeter {
         }
         info!(log, "starting oximeter server");
 
-        let resolver = Resolver::new_from_ip(
-            log.new(o!("component" => "DnsResolver")),
-            *args.address.ip(),
-        )?;
+        // Use the address for Oximeter to infer the bootstrap DNS address
+        let bootstrap_dns: Vec<SocketAddr> =
+            get_internal_dns_server_addresses(*args.address.ip())
+                .into_iter()
+                .map(|ip| SocketAddr::new(ip, DNS_PORT))
+                .collect();
+
+        let make_clickhouse_resolver = || -> BoxedResolver {
+            if let Some(address) = config.db.address {
+                Box::new(SingleHostResolver::new(address))
+            } else {
+                let service = if config.db.replicated {
+                    ServiceName::ClickhouseServer
+                } else {
+                    ServiceName::Clickhouse
+                };
+                Box::new(DnsResolver::new(
+                    service::Name(service.srv_name()),
+                    bootstrap_dns.clone(),
+                    DnsResolverConfig {
+                        hardcoded_ttl: Some(tokio::time::Duration::MAX),
```

Comment:
Is this right? It seems to imply we'll ignore the TTL in the DNS response and keep the first records indefinitely, though I could be misunderstanding.

Reply:
Yup! Right now, I *think* our internal DNS system is using a TTL of "zero", which means we'd see a backend, start to spin up connections for them, and then discard them immediately. Next time we query DNS, we'd start to create connections, then discard them immediately...

This behavior is tested here: https://github.com/oxidecomputer/qorb/blob/7c886a24369f458bdab4b9530e01f498da79e61f/src/resolvers/dns.rs#L712-L746 , but it sucks.

By ignoring TTLs, we still discard backends that are **explicitly** removed from DNS, but we only remove backends when DNS servers stop telling us about them successfully.

Reply:
We talked about this in chat. `qorb` independently refetches the list of backends from DNS periodically, and _also_ expires those backends using this TTL. The value of `MAX` here tells `qorb` to never expire them explicitly, but the list is still refreshed periodically when the query-interval expires. That all seems fine to me.